Milestone 3. Set up associations

indigodavid opened the pull request:
For this milestone, the following features have been implemented:

## New classes ##

- Classroom:
  - Attributes: label, students
  - Methods: add_student
- Book:
  - Attributes: title, author, rentals
- Rental:
  - Attributes: date, person, book

## Relationships ##

- Classroom `has-many` Students
- Student `belongs-to` a Classroom
- Person `has-many` Rentals
- Rental `belongs-to` a Person
- Book `has-many` Rentals
- Rental `belongs-to` a Book

indigodavid committed:
Add Classroom class

indigodavid committed:
Setup Classroom relationships

indigodavid committed:
Add Classroom object to Student class

indigodavid committed:
Add Book class

indigodavid committed:
Add Rental class

indigodavid committed:
Add many-to-many relationship between Book and Person using Rental class

indigodavid committed:
Fix linters errors

RolandM99 commented on lib/school_library/book.rb:
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
- [x] Instead of implementing rentals as an `attr_reader` I'll suggest you to use `attr_accessor`. This accessor will generate the automatic Getter & Setter method for the rentals item.

RolandM99 commented on lib/school_library/book.rb:
```diff
@@ -0,0 +1,10 @@
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
- [x] Kindly, make sure that you create a method to add rentals in your `Book` class. you can implement it as in the code snippet below:
```Ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end

```

RolandM99 commented on lib/school_library/person.rb:
```diff
@@ -2,14 +2,15 @@
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
- [x] Kindly, add the  `:rentals`  as an `attr_accessor` item.

RolandM99 commented on lib/school_library/person.rb:
```diff
@@ -2,14 +2,15 @@
     @parent_permission = parent_permission
     @id = Random.rand(1...100_000)
+    @rentals = []
   end
 
```
- [x] Kindly, make sure that you create a method to add rentals in your `Person` class. you can implement it as in the code snippet below:
```Ruby
 def add_rental(book, date)
    Rental.new(date, self, book)
  end

```

RolandM99 reviewed (changes requested):
Hi @indigodavid, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Gitflow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

indigodavid committed:
Fix Rental relationship with Book and Person

- Add attr_accessor for rentals on Book and Person classes
- Add method add_rental on Book and Person classes

mugishasam123 reviewed (approved):
Hi @indigodavid  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors 👍 
- Good commit messages ✔️ 
- Nice use of Git-flow, keep rocking 🚀 

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
